File: chat-service/src/main/java/com/relay/chatservice/user/UserService.java
package com.relay.chatservice.user;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    public void saveUser(User user) {

    }

    public void disconnect(User user) {

    }

    public List<User> findConnectedUsers() {
        return null;
    }
}

File: chat-service/src/main/java/com/relay/chatservice/user/UserService.kt
package com.relay.chatservice.user

import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class UserService(private val userRepository: UserRepository) {

    fun saveUser(user: User) {
        user.status = Status.ONLINE
        userRepository.save<User?>(user)
    }

    fun disconnect(user: User) {
        val storedUser = userRepository.findById(user.userName).orElse(null)
        if (storedUser != null) {
            storedUser.status = Status.OFFLINE
            userRepository.save<User?>(storedUser)
        }
    }

    fun findConnectedUsers(): MutableList<User?>? {
        return userRepository.findAllByStatus(Status.ONLINE)
    }
}
